### PR TITLE
Split management/public registration file upload

### PR DESCRIPTION
**Pull request by vtran99**
Request change to fix issue "Access denied" in private form when uploading file in registration from management area.

Tests done: upload file in front-end/back-end, with public and private form.

Notes:
I name the new RH for management similar to existing to be consistent, but in general the naming is a bit confusing to distinguish RH front/back-end.

### CODE REVIEW

File: indico/modules/events/registration/blueprint.py [hunk 1]
```diff
@@ -62,6 +62,8 @@
                  reglists.RHRegistrationCheckEmail)
 _bp.add_url_rule('/manage/registration/<int:reg_form_id>/registrations/delete', 'delete_registrations',
                  reglists.RHRegistrationDelete, methods=('POST',))
+_bp.add_url_rule('/manage/registration/<int:reg_form_id>/registrations/upload', 'registration_upload_file',
```
Comment (ThiefMaster) [inline indico/modules/events/registration/blueprint.py:65]: `upload_file_management` would be less confusing

File: indico/modules/files/controllers.py [hunk 2]
```diff
@@ -55,7 +62,7 @@ def get_file_context(self):
         return ``['event', EVENTID]`` so the file gets stored under
         ``event/EVENTID/...`` in the storage backend.
         """
-        raise NotImplementedError
+        return 'event', self.event.id, 'regform', self.regform.id, 'registration'
```
Comment (ThiefMaster) [inline indico/modules/files/controllers.py:65]: This particular mixin is quite generic and not only for registrations. You need to create a new mixin (in the registration module):

```py
class UploadRegistrationFileMixin(UploadFileMixin):
    @use_kwargs({
        'token': UUIDString(load_default=None),
    }, location='query')
    def _process_args(self, token):
        self.existing_registration = self.regform.get_registration(uuid=token) if token else None

    def get_file_context(self):
        return 'event', self.event.id, 'regform', self.regform.id, 'registration'
```

Then you can simply use that in the RHs.

**Comment by vtran99** [comment]
Hi Adrian,
When you have some time, could you please finalise this review ?
We received some complaints about this issue.
Thanks,

**Comment by ThiefMaster** [comment]
I added some cleanups (e.g. not having registration-specific stuff in the files module) and fixed the missing split for the picture upload RH.

**Pull request merged**